core/include/thread.h

```diff
@@ -102,13 +102,16 @@ struct _thread {
 #endif
 };
 
- /**
+/**
  * @def THREAD_STACKSIZE_DEFAULT
  * @brief A reasonable default stack size that will suffice most smaller tasks
  */
 #ifndef THREAD_STACKSIZE_DEFAULT
 #error THREAD_STACKSIZE_DEFAULT must be defined per CPU
 #endif
+#ifdef DOXYGEN
+#define THREAD_STACKSIZE_DEFAULT
```

Comment:
Yes, the `DOXYGEN` macro should be only set for doxygen (in https://github.com/RIOT-OS/RIOT/blob/master/doc/doxygen/riot.doxyfile#L1998). I don't know if there's any smarter way to do this.